### ISSUE TITLE
Video HLS stop playing when skip to certain timecode

### ISSUE THREAD

**Issue by deirpes** [videojs/video.js#6709]
## Description
Hello,
Currently, I have faced a big issue that when I click "play" and then skip to certain timecode, the video keeps loading and not playing anymore, I have created my sandbox and already attached my HLS link to video.js.
https://codesandbox.io/s/i2niy

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.
1. Click "play" button
2. Skip to certain timecode
3. See the results

### Expected
The video continues playing movie

### Actual
The video keeps loading. Although I check in the network tab, I see fragments continue downloading

### Error output
There is no error show in console

## Additional Information
Please include any additional information necessary here. Including the following:
#### videojs
Video.js v7.8.3
#### browsers
Google Chrome v83.0.4103.97
#### OSes
Mac OS High Sierra v10.13.6
### plugins
videojs-contrib-ads v6.6.5
videojs-ima v1.6.2

**Comment by gkatsev** [comment]
I took a quick look and it seems like it's something to do with the audio playlist. If I play any of the video playlists directly, it plays fine.
Playing back just the audio playlist plays back fine but seeking cause it to hiccup. However, seeking forward again causes it to play back.
Trying the main playlist again, after it gets stuck, if I seek forward a tiny bit again, it'll get unstuck.
This is likely because our choice for seeking isn't perfect, and we fail but then on a second seek we end up syncing correctly.

Running it through apple's media stream validator, there's a bunch of errors.

Here's the output
<details><code><pre>

--------------------------------------------------------------------------------
MUST fix issues
--------------------------------------------------------------------------------

Error: Playlist vs segment duration mismatch
--> Detail:  Segment duration 3.9938, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/34.ts

--> Detail:  Segment duration 4.0170, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/41.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8540
--> Source:  audio_0_1.m3u8 - audio_0_1/44.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/122.ts

--> Detail:  Segment duration 4.0171, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/154.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8540
--> Source:  audio_0_1.m3u8 - audio_0_1/211.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.6920
--> Source:  audio_0_1.m3u8 - audio_0_1/243.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8540
--> Source:  audio_0_1.m3u8 - audio_0_1/322.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/360.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.5060
--> Source:  audio_0_1.m3u8 - audio_0_1/398.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/400.ts

--> Detail:  Segment duration 4.0171, Playlist duration: 3.8780
--> Source:  audio_0_1.m3u8 - audio_0_1/402.ts

--> Detail:  Segment duration 3.9938, Playlist duration: 3.8310
--> Source:  audio_0_1.m3u8 - audio_0_1/404.ts

Error: Playlist video profile doesn't match video content profile
--> Detail:  Playlist: 0x4D, Video: 0x64
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_3.m3u8

--> Detail:  Playlist: 0x4D, Video: 0x64
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_4.m3u8

--> Detail:  Playlist: 0x4D, Video: 0x64
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_5.m3u8

--> Detail:  Playlist: 0x4D, Video: 0x64
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_2.m3u8

--> Detail:  Playlist: 0x4D, Video: 0x64
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_1.m3u8

Error: Playlist video resolution doesn't match parsed video resolution
--> Detail:  Playlist: 854 x 480, Video: 853 x 480
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_3.m3u8

Error: Measured peak bitrate compared to master playlist declared value exceeds error tolerance
--> Detail:  Measured: 1331.78 kb/s, Master playlist: 1567.00 kb/s, Error: 15.01%, Combined rendition name: eng
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_4.m3u8

--> Detail:  Measured: 803.50 kb/s, Master playlist: 1039.00 kb/s, Error: 22.67%, Combined rendition name: eng
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_5.m3u8

--> Detail:  Measured: 1972.86 kb/s, Master playlist: 2208.00 kb/s, Error: 10.65%, Combined rendition name: eng
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_3.m3u8

Error: Illegal MIME type
--> Detail:  MIME type: application/octet-stream
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8

--> Detail:  MIME type: application/octet-stream
--> Source:  video_3.m3u8

--> Detail:  MIME type: application/octet-stream
--> Source:  video_4.m3u8

--> Detail:  MIME type: application/octet-stream
--> Source:  video_5.m3u8

--> Detail:  MIME type: application/octet-stream
--> Source:  audio_0_1.m3u8

--> Detail:  MIME type: application/octet-stream
--> Source:  video_1.m3u8

--> Detail:  MIME type: application/octet-stream
--> Source:  video_2.m3u8

--------------------------------------------------------------------------------
SHOULD fix issues
--------------------------------------------------------------------------------

Warning: Playlist video level doesn't match video content level
--> Detail:  Playlist: 0x29, Video: 0x1E
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_3.m3u8

--> Detail:  Playlist: 0x29, Video: 0x1E
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_4.m3u8

--> Detail:  Playlist: 0x29, Video: 0x15
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_5.m3u8

--> Detail:  Playlist: 0x29, Video: 0x1F
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_2.m3u8

--> Detail:  Playlist: 0x29, Video: 0x28
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: video_1.m3u8

Warning: Non-canonical language subtag in language tag
--> Detail:  Unicode language: 'en', Language tag: 'eng'
--> Source:  https://s3.us-west-1.wasabisys.com/butacatest/hls/5a041044af1611ea880652f7f6f8cadc/playlist.m3u8
--> Compare: audio_0_1.m3u8
</pre></code></details>

**Comment by rgreen3400** [comment]
Thanks Gary...  I'm working with @deirpes on this.  The files were transcoded by qencode.com .  Perhaps we can try to encode with another service and see if we find better success as those stream errors appear to be from the transcode.  If not that's unfortunate on the videojs limitations on seek as seeking/scrubbing is pretty fundamental I would think and HLS is here to stay. thanks again.

**Comment by gkatsev** [comment]
There's only so much we can do to work with badly encoded streams. We can look at fixing this type of thing but it may be awhile before we can address it.

**Comment by rgreen3400** [comment]
Yes, absolutely.  So would you think we would have better success with a
service like Zencoder?

**Comment by gkatsev** [comment]
Zencoder should encode it properly, though, I have to say I'm a bit biased given Brightcove employs me.

**Comment by shenzekun** [comment]
+1
